### ISSUE TITLE
Under nginx https reverse proxy, base URI is http

### ISSUE THREAD

**Issue by scottdrichards** [firefly-iii/firefly-iii#3743]
I am running Firefly III version 5.3.3

**Description**
I am running firefly on a container and am connected to it via an nginx reverse proxy. I can load the registration page "https://firefly.example.com" however, some of the resources have links to "http://firefly.example.com" and the base URI in the registration page is "http..." (not https). The main error I get is: Refused to send form data to 'http://firefly.example.com/register' because it violates the following Content Security Policy directive: "form-action 'self'".

I assume this is just because I'm not ignorant with these systems (nginx, docker, etc.) so apologies if this isn't even a firefly issue. I'll add extra info if relevant, just mention it.

**Comment by JC5** [comment]
Some hints may be found in earlier issues. 

https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue+nginx

**Comment by scottdrichards** [comment]
Thank you, I had thought I had looked through past issues. It turns out I needed to change my nginx proxy config to include:
```
    proxy_set_header X-Forwarded-Host $host;
    proxy_set_header X-Forwarded-Proto $scheme;
```
The first line modifies the header to tell firefly the original destination of the request. "Hey, I know I'm sending this message to `localhost`, but the real request is being sent to `firefly.example.com`. [Explanation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host).
The second line does the same but for https vs http. [Explanation.](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto)
